spotinst_sdk2/models/ocean/aws/__init__.py

```diff
@@ -1,370 +1,918 @@
+    resource_label = "resource.label.${labelKey}"
+    namespace_annotation = "namespace.annotation.${annotationKey}"
+    resource_annotation = "resource.annotation.${annotationKey}"
+
+
+class AggregatedClusterCosts:
+    """
+    # Arguments
+    end_time: str
+    aggregated_filter: Filter
+    group_by: GroupBy
+    start_time: str
+    """
+    def __init__(
+            self,
+            end_time: str = none,
+            aggregated_filter: Filter = none,
+            group_by: GroupBy = GroupBy.namespace.value,
+            start_time: str = none):
+        self.end_time = end_time
+        self.aggregated_filter = aggregated_filter
+        self.group_by = group_by
+        self.start_time = start_time
 
-	def toJSON(self):
-		return json.dumps(self, default=lambda o: o.__dict__,
+
+class AggregatedClusterCostRequest:
+    def __init__(self, aggregated_cluster_costs: AggregatedClusterCosts = none):
+        self.end_time = aggregated_cluster_costs.end_time
+        self.start_time = aggregated_cluster_costs.start_time
+        self.group_by = aggregated_cluster_costs.group_by
+        self.filter = aggregated_cluster_costs.aggregated_filter
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__,
+                          sort_keys=True, indent=4)
+
+
+class Roll:
+    """
+    # Arguments
+    batch_min_healthy_percentage: int
+    batch_size_percentage: int
+    comment: str
+    disable_launch_spec_auto_scaling: bool
+    instance_ids: List[str]
+    launch_spec_ids: List[str]
+    respect_pdb: bool
+    """
+    def __init__(
+            self,
+            batch_min_healthy_percentage: int = none,
+            batch_size_percentage: int = none,
+            comment: str = none,
+            disable_launch_spec_auto_scaling: bool = none,
+            launch_spec_ids: List[str] = none,
+            instance_ids: List[str] = none,
+            respect_pdb: bool = none):
+        self.batch_min_healthy_percentage = batch_min_healthy_percentage
+        self.batch_size_percentage = batch_size_percentage
+        self.comment = comment
+        self.disable_launch_spec_auto_scaling = disable_launch_spec_auto_scaling
+        self.instance_ids = instance_ids
+        self.launch_spec_ids = launch_spec_ids
+        self.respect_pdb = respect_pdb
+
+
+class ClusterRollInitiateRequest:
+    def __init__(self, roll: Roll = none):
+        self.roll = roll
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__,
+                          sort_keys=True, indent=4)
+
+
+class Status:
+    """
+    # Arguments
+    status: str
+    """
+    def __init__(
+            self,
+            status: str = none):
+        self.status = status
+
+
+class UpdateRoll:
+    """
+    # Arguments
+    roll: Status
+    """
+    def __init__(
+            self,
+            roll: Status = none):
+        self.roll = roll
+
+
+class ClusterRollUpdateRequest:
```

Comment:
Do we really need all sub-classes here?

Reply:
Verified. These subclasses help create the required payload.